apps/protocol-frontend/src/components/ContributionsByDateShell.tsx

```diff
@@ -0,0 +1,37 @@
+import useContributionCountInRange from '../hooks/useContributionCount';
+import ContributionsByDateChart from './ContributionsByDateChart';
+import { TODAY_DATE, YEAR } from '../utils/constants';
+import { subWeeks, startOfDay, endOfDay } from 'date-fns';
+
+interface ContributionByDateShellProps {
+  guildIds: number[];
+}
+
+const ContributionByDateShell = ({
+  guildIds,
+}: ContributionByDateShellProps) => {
+  const {
+    isFetching,
+    isLoading,
+    isError,
+    data: contributionsByDate,
+  } = useContributionCountInRange({
+    id: null,
+    startDate: subWeeks(startOfDay(TODAY_DATE), YEAR),
+    endDate: endOfDay(TODAY_DATE),
+    guildIds,
+    excludeUnassigned: true,
+  });
+
+  return (
+    <ContributionsByDateChart
+      contributionsCount={contributionsByDate}
+      isFetching={isFetching}
+      isLoading={isLoading}
+      isError={isError}
+      daoId={guildIds[0]}
```

Comment:
what if guildIds is an empty array?

Reply:
I added some additional protection here for this, but it's coming from the query param and is set to an array to match the hook structure which is looking for an array of numbers since we use it in the user dashboard.

If someone doesnt enter the param in the url (feature/dao/ ) for example, the page is a 404 anyway so this won't be rendered

Reply:
Yes, but if this component is reused somewhere the developer might not be defensive in the higher level component. It is probably better to be defensive in this component because this line will error if an empty array is passed in.